### ISSUE TITLE
Accessibility Issues

### ISSUE THREAD

**Issue by dbollinger** [sanusart/react-dropdown-select#169]
I am testing the demo and docs pages with VoiceOver and macOS Big Sur, and finding some issues with keyboard controls and screen reader support.

1. Navigating options should make use of [aria-activedescendent](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-activedescendant_attribute) to support screen readers and other assistive technologies.  Without implementing aria-activedescendent, the assistive technology's cursor is not guaranteed to follow the browser's focus.
2. Pressing the `esc` key does not close the popup menu
3. When the cursor focus moves to the control, the screen reader does not read the current value(s) to the user
  1. There are also a number of issues for screen reader support in the "multi-select" variant that could probably be solved with some simple aria attributes.s

<img width="513" alt="image" src="https://user-images.githubusercontent.com/2576618/112851275-70e93280-9070-11eb-910c-2b82ced2619a.png">

<img width="788" alt="image" src="https://user-images.githubusercontent.com/2576618/112852354-75fab180-9071-11eb-8111-b36d04670c34.png">

I highly recommend adding an accessibility section to this library's README highlighting these issues and encouraging folks to help address them.  Additional guidance about best practices for combobox style controls can be found on w3: https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html.

**Comment by mitchdowney** [comment]
@sanusart I'm currently adding screen reader accessibility to our web site and noticed I'm having this issue with Mac VoiceOver as well. When group items are visually hovered, the screen reader is not announcing them. The issue can be reproduced on the demo page https://sanusart.github.io/react-dropdown-select/

Does anyone know how we can fix or work around this? I thought maybe I could listen for onHighlight events, read the title of the highlighted item, and set it as the `name` property for the dropdown...but it looks like there is only an onChange event and not an onHighlight event.

I tried manually setting aria-activedescendent to true through Dev Tools, but that isn't resulting in the item being read for me. Maybe that's not a valid test though.

**Comment by mitchdowney** [comment]
@sanusart hmm sorry, this might just be user error. I was of the impression elements on the page should only be navigated to with the Tab key, but I realized with Mac VoiceOver I can navigate into the dropdown menu by pressing ctrl + opt + left/right arrows.

It still seems like Tab would make more sense here since it is a menu of buttons...but maybe using ctrl + opt + left/right is a common UX for screen reader users?

**Comment by mitchdowney** [comment]
@sanusart @dbollinger I made some changes intended to improve screen reader accessibility (at least for Mac VoiceOver). These changes do not seem like a proper fix however, and I haven't tested it thoroughly across different screen readers. According to my research, it seems like using custom keyboard events are supposed to be avoided entirely to ensure proper custom dropdown accessibility...but that would require drastic re-writing, so I'm trying to at least partially improve screen reader handling (in my very limited testing).

I've also added a dropdownAriaDescription prop, since I think the user should be informed about the non-standard accessibility behavior of the dropdown.

I published these changes to npm as @podverse/react-dropdown-select in case someone else wants to use them. Any feedback would be appreciated. Thanks!

https://github.com/sanusart/react-dropdown-select/pull/232

**Comment by sanusart** [comment]
@mitchdowney thank you for investigation and PR draft. Will try to get to it asap. 
As far as I know, screen readers are not limited to tab keys only and have broad shortcuts. Saw in the docs of few of them.
